docs/buildx.md

```diff
@@ -0,0 +1,51 @@
+# Running Buildx
+
+Docker Buildx is a CLI plugin that extends the docker command with the full support of the features provided by Moby BuildKit builder toolkit. BuildKit is designed to work well for building for multiple platforms and not only for the architecture and operating system that the user invoking the build happens to run. More information can be found [docker-buildx](https://docs.docker.com/buildx/working-with-buildx/)
+
+### Enable docker experimental client flag
+```
+mkdir -p ~/.docker \
+    && echo "{\"experimental\": \"enabled\"}" >~/.docker/config.json \
+    && sudo service docker restart
+```
+
+On linux, `docker version` should show `Experimental: true` for Client Docker Engine
+
+For more information check [docker-arm](https://www.docker.com/blog/getting-started-with-docker-for-arm-on-linux/)
+
+### Enable docker experimental server flag
+```
+sudo mkdir -p /etc/systemd/system/docker.service.d && \
+	echo -e "[Service]\nExecStart=\nExecStart=/usr/bin/dockerd --experimental" | sudo tee -a /etc/systemd/system/docker.service.d/override.conf && \
+	sudo systemctl daemon-reload && \
+	sudo systemctl restart docker.service
+```
+On linux, `docker version` should show `Experimental: true` for Server Docker Engine
+
+### Enable execution of different multi-architecture QEMU
+```
+docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
+```
+
+For more information check [qemu-user-static](https://github.com/multiarch/qemu-user-static)
+
+### Create a build instance
+```
+docker buildx rm host-builder \
+    && docker buildx create --use --name host-builder --buildkitd-flags '--allow-insecure-entitlement network.host' \
+    && docker buildx use host-builder \
+    && docker buildx inspect --bootstrap
+```
+
+This creates a builder instance with name host-builder
+
+For more information check [docker-github](https://github.com/docker/buildx/blob/master/README.md)
+
+### Using buildx
+Docker login to enable push of images onto [dockerhub](https://hub.docker.com/u/cloudsuite/). 
+```
+docker login
+```
+
+#### Creating the base-os image
+Run `build.sh` script in `cloudsuite/commons/base-os` to generate base images for different architectures in one manifest
```

Comment:
When I building the images with the script, I get error for riscv image. 
here is the details about the error.
```
#7 79.63 W: Failure trying to run: chroot "/rootfs" /bin/true
#7 79.63 W: See /rootfs/debootstrap/debootstrap.log for details

failed to solve: rpc error: code = Unknown desc = executor failed running [/bin/sh -c set -ex     && echo "deb http://ftp.ports.debian.org/debian-ports sid main" >> /etc/apt/sources.list     && apt-get install -y debian-ports-archive-keyring     && dpkg --add-architecture riscv64     && apt-get update     && debootstrap     --keyring=/usr/share/keyrings/debian-ports-archive-keyring.gpg     --arch=riscv64     --variant=minbase     --include=debian-ports-archive-keyring     sid /rootfs http://ftp.ports.debian.org/debian-ports     && rm -rf /rootfs/var/log/dpkg.log /rootfs/var/log/bootstrap.log /rootfs/var/log/alternatives.log     && rm -rf /rootfs/var/cache/ldconfig/aux-cache /rootfs/var/cache/apt/*      && rm -rf /rootfs/var/lib/apt/lists/*]: buildkit-runc did not terminate successfully
```

Reply:
I tried to re-run on a new machine (ubuntu os, virtual machine), and the following command resolved the issue
```
docker run --rm --privileged multiarch/qemu-user-static --reset -p yes
```
I have already added this as part of readme for buildx